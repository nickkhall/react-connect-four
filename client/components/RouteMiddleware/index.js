import React from 'react';

// React router
import { Route } from 'react-router-dom';

export const renderRoute = (Component, routeProps) => <Component {...routeProps} />;

const RouteMiddleware = ({
  component: Component,
  ...routeProps
}) => (
  <Route
    {...routeProps}
    render={() => renderRoute(Component, routeProps)}
  />
);

export default RouteMiddleware;
