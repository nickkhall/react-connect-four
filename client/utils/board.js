export const getNextNum = (val, amount) => {
  const splitVal = val.split('');
  const nextLetter = String.fromCharCode((parseInt(splitVal[0].charCodeAt(0), 10) + amount));
  const nextNum = parseInt(splitVal[1], 10) - amount;

  return `${nextLetter}${nextNum}`;
};
