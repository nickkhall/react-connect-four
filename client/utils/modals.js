/**
 * Gets the active modal (if available) from the modal stack. In some situations
 * we have multiple active modals (such as when we open an image select or
 * vendor modal on top of an edit modal).
 * @param {!Array.<!Object>} stack The modal stack.
 * @return {{name: string, props: Object}} The current modal object.
 */
export const getActiveModal = stack => (stack && stack.length ? stack[stack.length - 1] : {});
