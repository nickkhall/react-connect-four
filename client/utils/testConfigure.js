/* eslint-disable import/no-extraneous-dependencies, no-console */
// React 16.x requires an Enzyme adapter.
// See: http://airbnb.io/enzyme/docs/installation/index.html#installation
// And: http://facebook.github.io/jest/docs/en/configuration.html#setuptestframeworkscriptfile-string
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

if (!global.TEST_CFG) {
  global.TEST_CFG = true;

  configure({ adapter: new Adapter() });

  const mockNoop = () => new Promise(() => {});

  jest.mock('axios', () => ({
    default: mockNoop,
    get: mockNoop,
    post: mockNoop,
    put: mockNoop,
    delete: mockNoop,
    patch: mockNoop
  }));

  jest.mock('store', () => ({
    dispatch: jest.fn()
  }));

  process.on('unhandledRejection', (e) => {
    console.error(e);
  });
}
