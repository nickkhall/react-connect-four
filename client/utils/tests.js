import React from 'react';

/* eslint-disable import/no-extraneous-dependencies */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

/* eslint-disable no-console */
export const mockNotificationDud = { type: 'MOCK_NOTIF_DUD' };

export const testingFunc = test => `should contain callable ${test} function`;

export const handleTestError = (test, err, done) => {
  console.warn('Error with', test);
  console.error(new Error(err));
  done();
};

/**
 * Automatically tests that the component matches the Jest snapshot.
 * NOTE: If you are receiving a warning like the following in your tests:
 *
 *   Warning: React.createElement: type is invalid -- expected a string...
 *
 * Then you most likely need to pass the appropriate props.
 *
 * @param {!React.Component} Component The React component to wrap and test.
 * @param {!Object} props The (optional) props to use for the basic Jest test.
 */
export const testCommonComponentAttrs = (Component, props) => {
  describe('when rendering', () => {
    const wrapper = shallow(<Component {...props} />);

    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
    });
  });
};

export const testDispatchToProps = (Component, mapDispatchToProps) => {
  let mockDispatch;
  let mockActions;

  beforeEach(() => {
    mockDispatch = jest.fn();
    mockActions = mapDispatchToProps(mockDispatch);
  });

  describe('when mapping dispatch to props', () => {
    const actions = Object.keys(mapDispatchToProps());
    actions.forEach((action) => {
      it(`should map the ${action} action`, () => {
        expect(mockActions[action]).toBeDefined();
        expect(typeof mockActions[action]).toEqual('function');
        mockActions[action]();
        expect(mockDispatch).toHaveBeenCalled();
      });
    });
  });
};

export const getComponentWrapper = (Component, props) => shallow(<Component {...props} />);


export const testReducerDefaults = (reducer, initialState) => {
  describe('Reducer Defaults', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(initialState);
    });
  });
};
