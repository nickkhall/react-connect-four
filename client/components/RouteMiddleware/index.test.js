// Test utils.
import { testCommonComponentAttrs } from 'utils/tests';

// Components
import RouteMiddleware from './index';

describe('<RouteMiddleware /> Component', () => {
  testCommonComponentAttrs(RouteMiddleware);
});
