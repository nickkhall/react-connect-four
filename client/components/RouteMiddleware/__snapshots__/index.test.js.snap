// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteMiddleware /> Component when rendering should render a component matching the snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
